prompb/remote.proto

```diff
@@ -43,3 +67,14 @@ message QueryResult {
   // Samples within a time series must be ordered by time.
   repeated prometheus.TimeSeries timeseries = 1;
 }
+
+// ChunkedReadResponse is a response when response_type equals STREAMED_XOR_CHUNKS.
+// We strictly stream full series after series, optionally split by time. This means that a single frame can contain
+// partition of the single series, but once a new series is started to be streamed it means that no more chunks will
+// be sent for previous one.
```

Comment:
Whilst this might be true, it seems like an unnecessary restriction that we might want to change in the future (esp for very long queries).  Should we consider not having this restriction as part of the interface?

Can we say anything about the order of the series we return?

Reply:
It's something that the other end can rely on for efficiency, and if there's a new TSDB implementation at some point where a different way makes sense we can always have a new type covering it. For now I'd say let users depend on this, rather than sacrificing this to an unknown future change.

Reply:
Hm. I can agree with @tomwilkie that it will change soon and then we will need to do deprecation process or different type which might be painful, but without that restriction as @brian-brazil says callers and receivers cannot assume that useful invariant and gain benefits which would have bad conequences on Prometheus PromQL current shape.

This means that I think I am bit more on restricting it as @brian-brazil suggests, but maybe it makes sense then be prepared or at least have an idea how we will do it if we will need to change that restriction.

Maybe leave the current shape and once new restriction come in, add some `enum SeriesOrder` to `ReadRequest`? In theory we can do that gradually once the different restriction arise? And keep it YAGNI for now.

Reply:
I'm still not seeing how we'd be changing this, and we can always extend the proto.